src/main/java/marquez/api/JobRun.java

```diff
@@ -1,56 +1,65 @@
+import java.util.*;
 
 public final class JobRun {
-  @NotNull private final Timestamp createdAt;
-  @NotNull private final UUID runGuid;
-  @NotNull private final List<String> runArgs;
-  @NotNull private final Timestamp startedAt;
-  @NotNull private final Timestamp endedAt;
-  @NotNull private final UUID jobVersionGuid;
-  @NotNull private final UUID inputDatasetVersionGuid;
-  @NotNull private final UUID outputDatasetVersionGuid;
-  @NotNull private final Timestamp latestHeartbeat;
-
-  @JsonCreator
+
+  @JsonIgnore private final UUID guid;
+  private final Timestamp createdAt;
+  private final Timestamp startedAt;
+  private final Timestamp endedAt;
+  private final UUID jobRunDefinitionGuid;
+  private final Integer currentState;
+
+  private static Map<JobRunState.State, Set<JobRunState.State>> validTransitions = new HashMap<>();
+
+  static {
+    validTransitions.put(
+        JobRunState.State.NEW,
+        new HashSet<JobRunState.State>() {
+          {
+            add(JobRunState.State.RUNNING);
+            add(JobRunState.State.ABORTED);
+          }
+        });
+    validTransitions.put(
+        JobRunState.State.RUNNING,
+        new HashSet<JobRunState.State>() {
+          {
+            add(JobRunState.State.COMPLETED);
+            add(JobRunState.State.FAILED);
+            add(JobRunState.State.ABORTED);
+          }
+        });
+    validTransitions.put(JobRunState.State.FAILED, new HashSet<>());
+    validTransitions.put(JobRunState.State.COMPLETED, new HashSet<>());
+    validTransitions.put(JobRunState.State.ABORTED, new HashSet<>());
+  }
+
   public JobRun(
-      @JsonProperty("createdAt") final Timestamp createdAt,
-      @JsonProperty("runGuid") final UUID runGuid,
-      @JsonProperty("runArgs") final List<String> runArgs,
-      @JsonProperty("startedAt") final Timestamp startedAt,
-      @JsonProperty("endedAt") final Timestamp endedAt,
-      @JsonProperty("jobVersionGuid") final UUID jobVersionGuid,
-      @JsonProperty("inputDatasetVersionGuid") final UUID inputDatasetVersionGuid,
-      @JsonProperty("outputDatasetVersionGuid") final UUID outputDatasetVersionGuid,
-      @JsonProperty("latestHeartbeat") final Timestamp latestHeartbeat) {
+      final UUID guid,
+      final Timestamp createdAt,
+      final Timestamp startedAt,
+      final Timestamp endedAt,
+      final UUID jobRunDefinitionGuid,
+      final Integer currentState) {
+    this.guid = guid;
     this.createdAt = createdAt;
-    this.runGuid = runGuid;
-    this.runArgs = runArgs;
     this.startedAt = startedAt;
     this.endedAt = endedAt;
-    this.jobVersionGuid = jobVersionGuid;
-    this.inputDatasetVersionGuid = inputDatasetVersionGuid;
-    this.outputDatasetVersionGuid = outputDatasetVersionGuid;
-    this.latestHeartbeat = latestHeartbeat;
+    this.jobRunDefinitionGuid = jobRunDefinitionGuid;
+    this.currentState = currentState;
   }
 
-  public Timestamp getCreatedAt() {
-    return createdAt;
+  @JsonIgnore
```

Comment:
TODO(code review): this won't be serialized into json, so json annotations not necessary.